pkg/storegateway/indexcache/cache_test.go

```diff
@@ -49,9 +50,64 @@ func BenchmarkCanonicalPostingsKey(b *testing.B) {
 	}
 	for numPostings := 10; numPostings <= len(ms); numPostings *= 10 {
 		b.Run(fmt.Sprintf("%d postings", numPostings), func(b *testing.B) {
+			b.ReportAllocs()
 			for i := 0; i < b.N; i++ {
 				_ = CanonicalPostingsKey(ms[:numPostings])
 			}
 		})
 	}
 }
+
+func TestUnsafeCastPostingsToBytes(t *testing.T) {
+	slowPostingsToBytes := func(postings []storage.SeriesRef) []byte {
+		byteSlice := make([]byte, len(postings)*8)
+		for i, posting := range postings {
+			for octet := 0; octet < 8; octet++ {
+				byteSlice[i*8+octet] = byte(posting >> (octet * 8))
+			}
+		}
+		return byteSlice
+	}
+	t.Run("base case", func(t *testing.T) {
+		postings := []storage.SeriesRef{1, 2}
+		assert.Equal(t, slowPostingsToBytes(postings), unsafeCastPostingsToBytes(postings))
+	})
+	t.Run("zero-length postings", func(t *testing.T) {
+		postings := make([]storage.SeriesRef, 0)
+		assert.Equal(t, slowPostingsToBytes(postings), unsafeCastPostingsToBytes(postings))
+	})
+	t.Run("nil postings", func(t *testing.T) {
+		assert.Equal(t, []byte(nil), unsafeCastPostingsToBytes(nil))
+	})
+	t.Run("more than 256 postings", func(t *testing.T) {
+		// Only casting a slice pointer truncates all postings to only their last byte.
+		postings := make([]storage.SeriesRef, 300)
+		for i := range postings {
+			postings[i] = storage.SeriesRef(i + 1)
+		}
+		assert.Equal(t, slowPostingsToBytes(postings), unsafeCastPostingsToBytes(postings))
+	})
+}
+
+func TestCanonicalPostingsKey(t *testing.T) {
```

Comment:
Could you add a test for the case where one list of postings is a sublist of the other? E.g.

```go
postings1 := []storage.SeriesRef{1, 2, 3, 4}
postings2 := []storage.SeriesRef{1, 2, 3, 4, 5}
```

Reply:
👍 added one in https://github.com/grafana/mimir/pull/3779/commits/0306cd72e267127b1a7d33ba6ca175774b01a8e8